Code samples generator: fix culture-specific printing of doubles

Decavoid opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Use invariant culture for printing floats and doubles in `generator/ServiceClientGeneratorLib/Example.cs`. This makes sure the generated content does not depend on the current culture (dot/comma as a decimal separator).

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open [issue][issues], please link to the issue here -->
When running `generator/ServiceClientGenerator/bin/Release/ServiceClientGenerator.exe` and the current culture uses comma as a decimal separator (e.g. French, fr-FR), the generated file `docgenerator/AWSSDKDocSamples/Lambda/Lambda.GeneratedSamples.cs`
becomes invalid, `0,7` is printed instead of `0.7`. This is the only file that is affected by this issue.

Valid output:
```csharp
RoutingConfig = new AliasRoutingConfiguration { AdditionalVersionWeights = new Dictionary<string, double> {
    { "1", 0.7 }
} }
```
Invalid output:
```csharp
RoutingConfig = new AliasRoutingConfiguration { AdditionalVersionWeights = new Dictionary<string, double> {
    { "1", 0,7 }
} }
```

## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
I added a unit test to `generator/ServiceClientGeneratorTests/UnitTests.cs`.

<!--- ## Screenshots (if appropriate) -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License
<!--- The SDK is released under the [Apache 2.0 license][license], so any code you submit will be released under that license -->
<!--- For substantial contributions, we may ask you to sign a [Contributor License Agreement (CLA)][cla] -->
<!--- Put an `x` in the below box if you confirm that this request can be released under the Apache 2 license -->
- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

dscpinheiro commented:
> I forked aws-sdk-net repo to add Unity WebGL support for the project I work on (that is for another PR). Unity WebGL doesn't support threads, so it [requires](https://issuetracker.unity3d.com/issues/webgl-task-awaited-with-task-dot-configureawait-false-does-not-continue-in-a-webgl-build) to change every `.ConfigureAwait(false)` to `.ConfigureAwait(true)`. I made this change in `generator/ServiceClientGeneratorLib/Generators/SourceFiles/BasePaginator.tt` template. Ran the generator. The unrelated file `docgenerator/AWSSDKDocSamples/Lambda/Lambda.GeneratedSamples.cs` appeared in `git diff` because current culture uses comma as a decimal point. This PR prevents that file from appearing in `git diff`.

Interesting. Thanks for sharing.

normj commented:
Out of curiosity and my lack of knowledge of Unity WebGL was the `.ConfigureAwait(<bool>)` the only change you had to make to get the SDK to work in WebGL?

Decavoid commented:
> Out of curiosity and my lack of knowledge of Unity WebGL was the `.ConfigureAwait(<bool>)` the only change you had to make to get the SDK to work in WebGL?

To make SDK support Unity WebGL I made these changes:

Added new `AWSConfigs.AsyncTasksContinueOnCapturedContext` bool property.

Added `ConfigureAwaitEx()` `Task` extension to AWS SDK:

```csharp
    public static ConfiguredTaskAwaitable ConfigureAwaitEx(this Task task)
    {
        return task.ConfigureAwait(AWSConfigs.AsyncTasksContinueOnCapturedContext);
    }
```

In `generator/ServiceClientGeneratorLib/Generators/SourceFiles/BasePaginator.tt` replaced `.ConfigureAwait(false)` with `.ConfigureAwaitEx()`. This requires the generator run.

Replaced `.ConfigureAwait(false)` with `.ConfigureAwaitEx()` in the following folders. I skipped AWS projects that I don't use.

- sdk/src/Core/*
- sdk/src/Services/CognitoIdentity/Custom/*
- sdk/src/Services/CognitoSync/Custom/*
- sdk/src/Services/SecurityToken/Custom/*

Ran generator, built AWS SDK, copied Core and services *.dll's to my project.

Changes in my project:

Instantiate AWS clients using the following pattern to avoid reading (Unity reads embedded files differently) `endpoints.json` file to resolve service URL:

```csharp
    public static Amazon***Client Create***Client(string id, string key, string region)
    {
        return new(id, key, new Amazon***Config()
        {
            ServiceURL = $"https://***.{region}.api.aws",
            AuthenticationRegion = region,
            DefaultConfigurationMode = DefaultConfigurationMode.Mobile,
        });
    }
```

Configured `AWSConfigs`:

```csharp
AWSConfigs.UseAlternateUserAgentHeader = true; // Unity doesn't allow changing User-Agent in browser

#if UNITY_WEBGL
        AWSConfigs.AsyncTasksContinueOnCapturedContext = true;
#else
        AWSConfigs.AsyncTasksContinueOnCapturedContext = false;
#endif
```

Set reasonable HTTP request timeout for all services:
```csharp
    ((DefaultConfiguration)Amazon***DefaultConfiguration.Mobile).TimeToFirstByteTimeout = Timespan.FromSeconds(***);
```

Created the following files from the corresponding AWS files replacing `System.Net.Http.HttpClient` with `UnityEngine.Networking.UnityWebRequest`:

- UnityHttpClientResponseData.cs (from HttpClientResponseData.cs)
- UnityHttpRequestMessageFactory.cs (from HttpRequestMessageFactory.cs)
- UnityHttpResponseMessageBody.cs (from HttpResponseMessageBody.cs)
- UnityHttpWebRequestMessage.cs (from HttpWebRequestMessage.cs)

Customized `RuntimePipeline` replacing default `HttpHandler<System.Net.Http.HttpContent>` with Unity-specific `HttpHandler<UnityEngine.Networking.UploadHandlerRaw>`:

```csharp
    RuntimePipelineCustomizerRegistry.Instance.Register(new UnityRuntimePipelineCustomizer());

    ...

    private class UnityRuntimePipelineCustomizer : IRuntimePipelineCustomizer
    {
        public string UniqueName => "UnityHttpHandler";

        public void Customize(Type type, RuntimePipeline pipeline)
        {
            var oldType = typeof(HttpHandler<System.Net.Http.HttpContent>);
            var oldHandler = (HttpHandler<System.Net.Http.HttpContent>)pipeline.Handlers.First(h => h.GetType() == oldType);
            AmazonServiceClient client = (AmazonServiceClient)oldHandler.CallbackSender;
            IClientConfig clientConfig = client.Config;

            UnityHttpRequestMessageFactory unityHttpRequestFactory = new(clientConfig);
            var unityHttpHandler = new HttpHandler<UnityEngine.Networking.UploadHandlerRaw>(unityHttpRequestFactory, this);

            pipeline.ReplaceHandler<HttpHandler<System.Net.Http.HttpContent>>(unityHttpHandler);
        }
    }
```

Created Unity GameObject with custom `UnityAwsSdk.cs` script that is responsible for running `UnityWebRequest`'s and running `Action`s on the Unity main thread.
